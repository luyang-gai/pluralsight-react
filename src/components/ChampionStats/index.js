import React from 'react';
import PropTypes from 'prop-types';
import Wrapper from './Wrapper';
import H4 from '../H4/index';
import ColoredText from '../ColoredText/index';

const ChampionStats = ({currentChampionStats}) => {
  const championWinPercent = Math.round(currentChampionStats.totalSessionsWon*100/currentChampionStats.totalSessionsPlayed);
  const averageKills = currentChampionStats.totalChampionKills / currentChampionStats.totalSessionsPlayed;
  const averageDeaths = currentChampionStats.totalDeathsPerSession / currentChampionStats.totalSessionsPlayed;
  const averageAssists = currentChampionStats.totalAssists / currentChampionStats.totalSessionsPlayed;
  const KDA = (averageKills + averageAssists) / averageDeaths;

  const getWinPercentColor = (percent) => {
    if (percent > 50) {
      return 'green';
    } else if (percent === 50) {
      return 'blue';
    } else {
      return 'red';
    }
  };

  const getKDAColor = (KDA) => {
    if (KDA > 4.5) {
      return 'green';
    } else if (KDA > 3) {
      return 'blue';
    } else {
      return 'red';
    }
  };

  return (
    <Wrapper>
      <H4>Champion stats</H4>
      <div>
        Winrate: <ColoredText color={getWinPercentColor(championWinPercent)}>{championWinPercent}%</ColoredText>
      </div>
      <div>Games played: {currentChampionStats.totalSessionsPlayed} games</div>
      <div>Record: {currentChampionStats.totalSessionsWon}W {currentChampionStats.totalSessionsLost}L</div>
      <div>KDA: <ColoredText color={getKDAColor(KDA)}>{KDA.toFixed(2)}</ColoredText></div>
      <div>{averageKills.toFixed(1)} K {averageDeaths.toFixed(1)} D {averageAssists.toFixed(1)} A</div>
    </Wrapper>
  );
};

/*
  EX:
  currentChampionStats: {

  }
 */
ChampionStats.propTypes = {
  currentChampionStats: PropTypes.object.isRequired
};

export default ChampionStats;
