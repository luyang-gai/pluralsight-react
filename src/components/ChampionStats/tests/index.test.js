import React from 'react';
import ChampionStats from '../index';
import renderer from 'react-test-renderer';
import mockChampionStats from '../../../mock/CurrentChampionStats';

describe('<ChampionStats/>', () => {
  it('renders basic ChampionStats correctly', () => {
    const tree = renderer.create(
      <ChampionStats currentChampionStats={mockChampionStats}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
