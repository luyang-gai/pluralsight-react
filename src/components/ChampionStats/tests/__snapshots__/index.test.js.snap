// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChampionStats/> renders basic ChampionStats correctly 1`] = `
<div
  className="foqgVb"
>
  <h4
    className="wkvwR"
  >
    Champion stats
  </h4>
  <div>
    Winrate: 
    <span
      className="gMHEVS"
      color="green"
    >
      57
      %
    </span>
  </div>
  <div>
    Games played: 
    46
     games
  </div>
  <div>
    Record: 
    26
    W 
    20
    L
  </div>
  <div>
    KDA: 
    <span
      className="YsgVl"
      color="red"
    >
      2.88
    </span>
  </div>
  <div>
    8.1
     K 
    5.8
     D 
    8.7
     A
  </div>
</div>
`;
